packages/core/src/components/DataTable/DataTable.tsx

```diff
@@ -224,10 +222,11 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
 
     if (dynamicRowHeight) {
       // We need to make sure the cache is cleared before React tries to re-render.
-      window.setTimeout(() => {
+      if (this.timeoutId) window.clearTimeout(this.timeoutId);
```

Comment:
If a timer is already set, shouldn't that one just run instead of clearing and creating a new one?